src/screens/DiscoveryScreenV2/hooks/useDiscovery.js

```diff
@@ -70,13 +74,37 @@ const useDiscovery = () => {
 
   const updateFollowDiscoveryContext = (willFollow, item) => {
     const newFollowedUsers = [...users];
-    const newUser = exhangeFollower(
-      newFollowedUsers,
+    const userId = item.user ? item.user.user_id : item.user_id;
+
+    // Update initial users
+    const newInitialUser = exchangeFollower(newFollowedUsers, willFollow, userId);
+    console.log('newInitialUser', !!newInitialUser);
+    if (newInitialUser !== null)
+      DiscoveryAction.setDiscoveryInitialUsers(
+        mapUserWith([...discovery?.initialUsers], newInitialUser),
+        discoveryDispatch
+      );
+
+    // Update discovery search users followed section
+    const newSearchFollowedUser = exchangeFollower(discovery?.followedUsers, willFollow, userId);
+    if (newSearchFollowedUser !== null)
+      DiscoveryAction.setNewFollowedUsers(
+        mapUserWith([...discovery?.followedUsers], newSearchFollowedUser),
+        discoveryDispatch
+      );
+
+    // Update discovery search users unfollowed section
+    const newSearchUnfollowedUser = exchangeFollower(
+      discovery?.unfollowedUsers,
       willFollow,
-      item.user ? item.user.user_id : item.user_id
+      userId
     );
-
-    DiscoveryAction.setDiscoveryInitialUsers(mapUser(newUser), discoveryDispatch);
+    console.log('newSearchUnfollowedUser', !!newSearchUnfollowedUser);
+    if (newSearchUnfollowedUser !== null)
+      DiscoveryAction.setNewUnfollowedUsers(
+        mapUserWith([...discovery?.unfollowedUsers], newSearchUnfollowedUser),
+        discoveryDispatch
+      );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This section has been significantly refactored to improve data handling and performance. The use of `exchangeFollower` and `mapUserWith` functions helps to reduce redundancy and improve code maintainability. However, there are console.log statements at lines 81 and 102 which should be removed to avoid unnecessary logging in production.



<!-- This is an auto-generated comment by OSS CodeRabbit -->